spec/DefinedSchemas.spec.js

```diff
@@ -0,0 +1,659 @@
+        addField: {},
+        protectedFields: { '*': ['aField'], 'role:Admin': ['anotherField'] },
+      };
+      const schemas = {
+        definitions: [
+          {
+            className: 'Test',
+            fields: { aField: { type: 'String' }, anotherField: { type: 'Object' } },
+            classLevelPermissions: expectedTestCLP,
+          },
+        ],
+      };
+      await new DefinedSchemas(schemas, server.config).execute();
+
+      let testSchema = await new Parse.Schema('Test').get();
+      expect(testSchema.classLevelPermissions).toEqual(expectedTestCLP);
+
+      expectedTestCLP.update = {};
+      expectedTestCLP.create = { requiresAuthentication: true };
+
+      await new DefinedSchemas(schemas, server.config).execute();
+      testSchema = await new Parse.Schema('Test').get();
+      expect(testSchema.classLevelPermissions).toEqual(expectedTestCLP);
+    });
+    it('should force addField to empty', async () => {
+      const server = await reconfigureServer();
+      const schemas = {
+        definitions: [{ className: 'Test', classLevelPermissions: { addField: { '*': true } } }],
+      };
+      await new DefinedSchemas(schemas, server.config).execute();
+
+      const expectedTestCLP = {
+        find: {},
+        count: {},
+        get: {},
+        create: {},
+        update: {},
+        delete: {},
+        addField: {},
+        protectedFields: {},
+      };
+
+      let testSchema = await new Parse.Schema('Test').get();
+      expect(testSchema.classLevelPermissions).toEqual(expectedTestCLP);
+
+      await new DefinedSchemas(schemas, server.config).execute();
+      testSchema = await new Parse.Schema('Test').get();
+      expect(testSchema.classLevelPermissions).toEqual(expectedTestCLP);
+    });
+  });
+
+  it('should not delete automatically classes', async () => {
+    await reconfigureServer({
+      schema: { definitions: [{ className: '_User' }, { className: 'Test' }] },
+    });
+
+    await reconfigureServer({ schema: { definitions: [{ className: '_User' }] } });
+
+    const schema = await new Parse.Schema('Test').get();
+    expect(schema.className).toEqual('Test');
+  });
+
+  it('should disable class PUT/POST endpoint when lockSchemas provided to avoid dual source of truth', async () => {
+    await reconfigureServer({
+      schema: {
+        lockSchemas: true,
+        definitions: [{ className: '_User' }, { className: 'Test' }],
+      },
+    });
+
+    const schema = await new Parse.Schema('Test').get();
+    expect(schema.className).toEqual('Test');
+
+    const schemas = await Parse.Schema.all();
+    expect(schemas.length).toEqual(3);
+
+    try {
+      await new Parse.Schema('TheNewTest').save();
+      fail('TheNewTest.save() should have failed');
+    } catch (e) {
+      expect(e.message).toContain('Cannot perform this operation when schemas options is used.');
+    }
+
+    try {
+      await new Parse.Schema('_User').update();
```

Comment:
Prefer `await expectAsync().toBeRejectedWith()`

Reply:
on it